swan-cmd/src/main/java/de/fraunhofer/iem/swan/io/dataset/InvokedMethodsFilter.java

```diff
@@ -0,0 +1,57 @@
+package de.fraunhofer.iem.swan.io.dataset;
+
+import de.fraunhofer.iem.swan.data.Method;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+/**
+ * Filters invoked methods in the application that are already part of the dataset
+ *
+ */
+
+public class InvokedMethodsFilter {
+
+    Set<Method> testMethods;
+    static Set<Method> invokedMethodsInDataset;
+    Set<Method> filteredTestMethods;
+    Set<Method> datasetMethods;
+    private static final Logger logger = LoggerFactory.getLogger(InvokedMethodsFilter.class);
+
+    public InvokedMethodsFilter(Set<Method> testMethods, Set<Method> datasetMethods){
+        this.testMethods = testMethods;
+        this.datasetMethods = datasetMethods;
+        invokedMethodsInDataset = new HashSet<>();
+        this.filteredTestMethods = new HashSet<>();
+    }
+
+    public Set<Method> filterUnknownInvokedMethods(){
+        for(Method method : testMethods){
+            if(method.isApplicationMethod()){
+                filteredTestMethods.add(method);
+                continue;
+            }
+            boolean inDataset = false;
+            for(Method datasetMethod : datasetMethods){
+                String testMethodSignature = method.getTrimmedSignature();
+                if(datasetMethod.getTrimmedSignature().equals((testMethodSignature))){
+                    method.setKnown(true);
+                    logger.info("Invoked method: {} is part of the dataset.", method.getTrimmedSignature());
+                    invokedMethodsInDataset.add(method);
```

Comment:
Wouldn't this add the method object from the test set which doesn't have any SRM/CWE and other properties to SWAN's output? So the datasetMethod instead should be used instead of method. Or am I missing something?

Reply:
Yes. This is a bug and has been fixed.